Build with Clang failed

When building the library with Clang instead of gcc, following errors are raised.

``` bash
/sick/sick_safetyscanners_base/src/SickSafetyscanners.cpp:54:7: error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
      std::move(sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port)))
      ^
/sick/sick_safetyscanners_base/src/SickSafetyscanners.cpp:54:7: note: remove std::move call here
      std::move(sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port)))
```

According to the documentation  :
> This warning warns when a call to std::move prevents copy elision. A typical scenario when copy elision can occur is when returning in a function with a class return type, when the expression being returned is the name of a non-volatile automatic object, and is not a function parameter, and has the same type as the function return type.

--- a/include/sick_safetyscanners_base/SickSafetyscanners.h
+++ b/include/sick_safetyscanners_base/SickSafetyscanners.h
@@ -62,7 +62,6 @@
 #include "sick_safetyscanners_base/cola2/Command.h"
 #include "sick_safetyscanners_base/datastructure/Datastructure.h"
 
-
 namespace sick {
 
 using io_service_ptr = std::shared_ptr<boost::asio::io_service>;
@@ -262,9 +261,7 @@ public:
 
 private:
   sick::types::ip_address_t m_sensor_ip;
-  sick::types::port_t m_sensor_tcp_port;
   CommSettings m_comm_settings;
-  bool m_is_initialized;
   std::unique_ptr<boost::asio::io_service> m_io_service_ptr;
 
   /*!

--- a/src/SickSafetyscanners.cpp
+++ b/src/SickSafetyscanners.cpp
@@ -45,13 +45,12 @@ SickSafetyscannersBase::SickSafetyscannersBase(sick::types::ip_address_t sensor_
                                                sick::types::port_t sensor_tcp_port,
                                                CommSettings comm_settings)
   : m_sensor_ip(sensor_ip)
-  , m_sensor_tcp_port(sensor_tcp_port)
   , m_comm_settings(comm_settings)
   , m_io_service_ptr(sick::make_unique<boost::asio::io_service>())
   , m_io_service(*m_io_service_ptr)
   , m_udp_client(m_io_service, comm_settings.host_udp_port)
   , m_session(
-      std::move(sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port)))
+      sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port))
   , m_packet_merger()
 {
   changeSensorSettings(comm_settings);
@@ -62,13 +61,12 @@ SickSafetyscannersBase::SickSafetyscannersBase(sick::types::ip_address_t sensor_
                                                CommSettings comm_settings,
                                                boost::asio::ip::address_v4 interface_ip)
   : m_sensor_ip(sensor_ip)
-  , m_sensor_tcp_port(sensor_tcp_port)
   , m_comm_settings(comm_settings)
   , m_io_service_ptr(sick::make_unique<boost::asio::io_service>())
   , m_io_service(*m_io_service_ptr)
   , m_udp_client(m_io_service, comm_settings.host_udp_port, comm_settings.host_ip, interface_ip)
   , m_session(
-      std::move(sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port)))
+      sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port))
   , m_packet_merger()
 {
   changeSensorSettings(comm_settings);
@@ -79,13 +77,12 @@ SickSafetyscannersBase::SickSafetyscannersBase(sick::types::ip_address_t sensor_
                                                CommSettings comm_settings,
                                                boost::asio::io_service& io_service)
   : m_sensor_ip(sensor_ip)
-  , m_sensor_tcp_port(sensor_tcp_port)
   , m_comm_settings(comm_settings)
   , m_io_service_ptr(nullptr)
   , m_io_service(io_service)
   , m_udp_client(m_io_service, comm_settings.host_udp_port)
   , m_session(
-      std::move(sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port)))
+      sick::make_unique<sick::communication::TCPClient>(m_sensor_ip, sensor_tcp_port))
   , m_packet_merger()
 {
   changeSensorSettings(comm_settings);

